frontend/packages/db-structure/src/parser/schemarb/parser.ts

```diff
@@ -581,6 +596,180 @@ function extractCheckConstraintWithTableName(
+ */
+function processInterleaveKeywordHashNode(
+  hashNode: KeywordHashNode,
+  interleaveConstraint: InterleaveConstraint,
+): void {
+  for (const argElement of hashNode.elements) {
+    if (!(argElement instanceof AssocNode)) continue
+
+    // @ts-expect-error: unescaped is defined as string but it is actually object
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    const key = argElement.key.unescaped.value
+    const value = argElement.value
+
+    processInterleaveOption(key, value, interleaveConstraint)
+  }
+}
+
+/**
+ * Process a single option for Spanner interleave constraint
+ *
+ * Handles specific option keys:
+ * - `column`: Sets the foreign key column name
+ * - `primary_key`: Sets the target primary key column name
+ * - `name`: Sets the constraint name
+ * - `on_delete`: Sets the delete constraint action (CASCADE or NO_ACTION)
+ *
+ * @param key - The option key (e.g., 'column', 'name', 'on_delete')
+ * @param value - The AST node containing the option value
+ * @param interleaveConstraint - The constraint object to modify
+ *
+ * @see {@link https://github.com/googleapis/ruby-spanner-activerecord/blob/a7ca729dc6fe13b1b23fda6698abab9e2c1ace8e/examples/snippets/interleaved-tables/README.md#models-for-interleaved-tables}
+ */
+function processInterleaveOption(
+  key: string,
+  value: Node,
+  interleaveConstraint: InterleaveConstraint,
+): void {
+  switch (key) {
+    case 'column':
+      if (value instanceof StringNode || value instanceof SymbolNode) {
+        interleaveConstraint.columnName = value.unescaped.value
+      }
+      break
+    case 'primary_key':
+      if (value instanceof StringNode || value instanceof SymbolNode) {
+        interleaveConstraint.targetColumnName = value.unescaped.value
+      }
+      break
+    case 'name':
+      if (value instanceof StringNode || value instanceof SymbolNode) {
+        interleaveConstraint.name = value.unescaped.value
+      }
+      break
+    case 'on_delete':
+      if (value instanceof SymbolNode) {
+        const deleteConstraint = normalizeInterleaveConstraintName(
+          value.unescaped.value,
+        )
+        interleaveConstraint.deleteConstraint = deleteConstraint
+      }
+      break
+  }
+}
+
+/**
+ * Normalize interleave constraint delete action value
+ *
+ * Google Cloud Spanner interleave constraints only support CASCADE and NO_ACTION
+ * for delete operations. Update operations are always NO_ACTION.
+ *
+ * @param constraint - The raw constraint string from Ruby (e.g., 'cascade')
+ * @returns Normalized constraint action ('CASCADE' or 'NO_ACTION')
+ */
+function normalizeInterleaveConstraintName(
+  constraint: string,
+): 'CASCADE' | 'NO_ACTION' {
+  // For interleave constraints, only CASCADE and NO_ACTION are valid
+  switch (constraint) {
+    case 'cascade':
+      return 'CASCADE'
+    default:
+      return 'NO_ACTION'
+  }
+}
```

Comment:
[memo]
These are based on foreign key constraint implementation.
The main difference is action type (CASCADE, NO_ACTION only)